backend/GymBeam/Controllers/ReservationController.cs

```diff
@@ -39,12 +43,64 @@ public async Task<ActionResult<IEnumerable<ReservationResponse>>> Get()
         }
 
         [HttpPost]
+        [ProducesResponseType(StatusCodes.Status204NoContent)]
+        [ProducesResponseType(StatusCodes.Status400BadRequest)]
+        [ProducesResponseType(StatusCodes.Status404NotFound)]
+        [ProducesResponseType(StatusCodes.Status405MethodNotAllowed)]
+        [ProducesResponseType(StatusCodes.Status500InternalServerError)]
 #if !DEBUG
         [Authorize(Roles = Roles.User)]
 #endif
-        public IActionResult Post([FromBody] ReservationRequest dto)
+        public async Task<ActionResult> Post([FromBody] ReservationRequest dto)
         {
-            return NoContent();
+            var request = new CreateReservationCommand
+            {
+                ActivityId = dto.ActivityId,
+                UserId = dto.UserId,
+                StartTime = dto.StartTime
+            };
+            try
+            {
+                if (!Request.Cookies.TryGetValue(Cookies.UserId, out string cookiesUserId))
+                    throw new InvalidCookieException(Cookies.UserId);
+                if (!int.TryParse(cookiesUserId, out int userIdFromCookies))
+                    throw new InvalidUserIdException();
+
+                request.LoggedUserId = userIdFromCookies;
```

Comment:
maybe we can retrive this logged user inside command handler like you did with claim role? If not possible then ok.